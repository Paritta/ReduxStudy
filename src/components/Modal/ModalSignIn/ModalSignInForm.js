import React from "react";
import styled from "styled-components";
import { Field, reduxForm } from "redux-form";
import oc from "open-color";
import FaGoogle from "react-icons/lib/fa/google";
import FaFacebook from "react-icons/lib/fa/facebook";
import FaTwitter from "react-icons/lib/fa/twitter";
import PropTypes from "prop-types";

const propTypes = {
    handleSubmit: PropTypes.func,
    hideModal: PropTypes.func,
    googleLogin: PropTypes.func,
};

const defaultTypes = {
    handleSubmit() {},
    hideModal() {},
    googleLogin() {},
};

const Wrapper = styled.div`
    padding: 30px 0 10px 0;
`;

const StyledField = styled(Field)`
    background-color: ${oc.gray[0]};
    width: 400px;
    height: 40px;
    font-size: 15px;
    border: 2px solid white;
    padding-left: 20px;
    margin-bottom: 10px;
`;

const Button = styled.button`
    background: palevioletred;
    color: white;
    
    font-size: 1em;
    margin: 1em;
    padding: 0.25em 1em;
    
    border: 2px solid palevioletred;
    border-radius: 3px;
    
    transition: all 0.3s ease;
    
    &:hover {
        background: ${oc.pink[4]};
        border: 2px solid ${oc.pink[4]};
    }
`;

const SocialLoginButton = styled.button`
    color: white;
    
    font-size: 1em;
    margin: 5px 0;
    padding: 0.5em 12em 0.5em 1em;
    
    transition: all 0.3s ease;
`;

const GoogleSocialLoginButton = SocialLoginButton.extend`
    background: red;
    border: 2px solid red;
    
    &:hover {
        background: ${oc.red[4]};
        border: 2px solid ${oc.red[4]};
    }
`;

const SocialLoginButtonWrapper = styled.div`
    padding: 10px 60px ;
    margin: 0;
`;

const Span = styled.span`
    padding: 0;
    margin: 0 0 0 15px;
`;


const FacebookSocialLoginButton = SocialLoginButton.extend`
    background: royalblue;
    border: 2px solid royalblue;
    padding: 0.5em 11.3em 0.5em 0.5em;
    
    &:hover {
        background: ${oc.blue[4]};
        border: 2px solid ${oc.blue[4]};
    }
`;

const TwitterSocialLoginButton = SocialLoginButton.extend`
    background: skyblue;
    border: 2px solid skyblue;
    
    &:hover {
        background: ${oc.blue[2]};
        border: 2px solid ${oc.blue[2]};
    }
`;

export let ModalSignInForm = ({ handleSubmit, hideModal, googleLogin }) => {
    return (
        <Wrapper>
            <form onSubmit={ handleSubmit }>
                <div>
                    <StyledField name="email" component="input" type="text" required placeholder="email"/>
                </div>
                <div>
                    <StyledField name="password" component="input" type="password" required placeholder="password"/>
                </div>
                <Button type="submit">로그인</Button>
                <Button onClick={() => hideModal()}>나가기</Button>
            </form>
            <SocialLoginButtonWrapper>
                <GoogleSocialLoginButton
                    onClick={() => googleLogin()}
                >
                    <FaGoogle />
                    <Span>Sign in with Google</Span>
                </GoogleSocialLoginButton>
                <FacebookSocialLoginButton>
                    <FaFacebook />
                    <Span>Sign in with Facebook</Span>
                </FacebookSocialLoginButton>
                <TwitterSocialLoginButton>
                    <FaTwitter />
                    <Span>Sign in with Twitter</Span>
                </TwitterSocialLoginButton>
            </SocialLoginButtonWrapper>
        </Wrapper>
    )
};

ModalSignInForm.propTypes = propTypes;
ModalSignInForm.defaultTypes = defaultTypes;

ModalSignInForm = reduxForm({
    form: "ModalSignIn"
})(ModalSignInForm);

export default ModalSignInForm;