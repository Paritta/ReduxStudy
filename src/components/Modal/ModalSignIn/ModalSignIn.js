import React from "react";
import styled, {keyframes} from "styled-components";
import { bounceInLeft, bounceOutRight } from 'react-animations';
import PropTypes from "prop-types";
import ModalSignInForm from "./ModalSignInForm";
import { pathToJS, firebaseConnect } from "react-redux-firebase";
import { connect } from "react-redux";
import ReactLoading from "react-loading";

const propTypes = {
    Modal: PropTypes.object,
    hideModal: PropTypes.func,
    Animate: PropTypes.object,
    AnimateTurn: PropTypes.func,
    AnimateDown: PropTypes.func,
    firebase: PropTypes.object,
    fetchRequest: PropTypes.func,
    toppopuReceiveRequest: PropTypes.func,
};

const defaultTypes = {
    Modal: {},
    hideModal() {},
    Animate: {},
    AnimateTurn() {},
    AnimateDown() {},
    fetchRequest() {},
    toppopuReceiveRequest() {},
};

const bounceInLeftAnimation = keyframes`${bounceInLeft}`;
const bounceOutRightAnimation = keyframes`${bounceOutRight}`;

const AnimationWrapper = styled.div`
    animation: 0.5s ${bounceInLeftAnimation};
`;

const AnimationOutWrapper = styled.div`
    animation: 0.5s ${props => props.AnimateOut ? bounceOutRightAnimation : ""};
`;


const Wrapper = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    
    overflow: hidden;
    
    z-index: 10;
    
    width: 550px;
`;

const ModalHeader = styled.div`
    font-style: normal;
    font-weight: 400;
    color: white;
    padding: 15px 15px;
    background: palevioletred;
`;

const ModalBox = styled.div`
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.3);
    padding: 15px;
`;

const Dimmed = styled.div`
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    
    z-index: 5;
    
    background: rgba(0, 0, 0, 0.3);
`;

const WrappedReactLoading = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    transform: scale(2);
    padding: 80px 0;
`;

export class ModalSignIn extends React.Component {
    state = {
      isLoading: false,
    };

    submit = values => {
        this.Login(values);
    };

    Login = LoginData => {
        this.setState({ isLoading: true });
        return this.props.firebase
            .login({
                email: LoginData.email,
                password: LoginData.password
            })
            .then(() => {
                this.setState({ isLoading: false });
                this.props.hideModal();
                this.props.fetchRequest();
                this.props.toppopuReceiveRequest();
            })
            .catch((error) => {
                this.setState({ isLoading: false });
                console.log("there was an error:", error);
                console.log("error prop:", this.props.authError);
            })
    };

    // 구글 로그인
    googleLogin = loginData => {
        this.setState({ isLoading: true });
        return this.props.firebase
            .login({ provider: 'google' })
            .then(() => {
                this.setState({ isLoading: false });
                // this is where you can redirect to another route
                this.props.fetchRequest();
                this.props.toppopuReceiveRequest();
            })
            .catch((error) => {
                this.setState({ isLoading: false });
                console.log('there was an error', error);
                console.log('error prop:', this.props.authError) // thanks to connect
            })
    };

    // 페이스북 로그인
    facebookLogin = loginData => {
        this.setState({ isLoading: true });
        return this.props.firebase
            .login({ provider: 'facebook' })
            .then(() => {
                this.setState({ isLoading: false });
                // this is where you can redirect to another route
                this.props.fetchRequest();
                this.props.toppopuReceiveRequest();
            })
            .catch((error) => {
                this.setState({ isLoading: false });
                console.log('there was an error', error);
                console.log('error prop:', this.props.authError) // thanks to connect
            })
    };

    render () {
        const { hideModal, animateTurn, animateDown, Animate } = this.props;

        return (
            <div>
                <Dimmed />
                <AnimationOutWrapper
                    AnimateOut={Animate.AnimateOut}
                >
                    <AnimationWrapper>
                        <Wrapper>
                                <ModalHeader>로그인</ModalHeader>
                                <ModalBox>
                                {!this.state.isLoading?
                                        <ModalSignInForm
                                            hideModal={hideModal}
                                            animateTurn={animateTurn}
                                            animateDown={animateDown}
                                            googleLogin={this.googleLogin}
                                            facebookLogin={this.facebookLogin}
                                            onSubmit={this.submit}
                                        />
                                    :
                                    <WrappedReactLoading>
                                        <ReactLoading type="cylon" color="palevioletred"/>
                                    </WrappedReactLoading>
                                }
                                </ModalBox>
                        </Wrapper>
                    </AnimationWrapper>
                </AnimationOutWrapper>
            </div>
        )
    }
}

ModalSignIn.propTypes = propTypes;
ModalSignIn.defaultTypes = defaultTypes;

const WrappedModalSignIn = firebaseConnect()(ModalSignIn);

export default connect(
    ({ firebase }) => ({
        authError: pathToJS(firebase, "authError"),
    })
)(WrappedModalSignIn);
