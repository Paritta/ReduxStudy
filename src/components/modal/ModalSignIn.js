import React from "react";
import styled, {keyframes} from "styled-components";
import { fadeIn } from 'react-animations';
import PropTypes from "prop-types";
import ModalSignInForm from "./ModalSignInForm";

const propTypes = {
    Modal: PropTypes.object,
    hideModal: PropTypes.func,
};

const defaultTypes = {
    hideModal() {},
};

const fadeInAnimation = keyframes`${fadeIn}`;

const Wrapper = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    
    overflow: hidden;
    
    z-index: 10;
    
    width: 550px;
    
    animation: 1s ${fadeInAnimation};
`;

const AnimationWrapper = styled.div`
    animation: 1s ${fadeInAnimation};
`;

const ModalHeader = styled.div`
    @import url(//fonts.googleapis.com/earlyaccess/hanna.css);
    font-family: 'Hanna', fantasy;
    font-style: normal;
    font-weight: 400;
    color: white;
    padding: 15px 15px;
    background: palevioletred;
`;

const ModalBox = styled.div`
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.3);
    padding: 15px;
`;

const Dimmed = styled.div`
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    
    z-index: 5;
    
    background: rgba(0, 0, 0, 0.3);
`;

class ModalSignIn extends React.Component {
    submit = values => {
        console.log(values);
    };

    render () {
        const { hideModal } = this.props;

        return (
            <div>
                <Dimmed />
                <Wrapper>
                    <AnimationWrapper>
                        <ModalHeader>로그인</ModalHeader>
                        <ModalBox>
                            <ModalSignInForm
                                hideModal={hideModal}
                                onSubmit={this.submit}
                            />
                        </ModalBox>
                    </AnimationWrapper>
                </Wrapper>
            </div>
        )
    }
};

ModalSignIn.propTypes = propTypes;
ModalSignIn.defaultTypes = defaultTypes;

export default ModalSignIn;
