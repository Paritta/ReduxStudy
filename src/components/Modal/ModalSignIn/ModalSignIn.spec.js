import React from "react";
import ReactTestRenderer from "react-test-renderer"
import { ModalSignIn } from "./ModalSignIn";
import "jest-styled-components";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {};
const store = mockStore(initialState);

it("ModalSignIn render correctly", () => {
    const Animate = {
        AnimateOut: ""
    };

    const tree = ReactTestRenderer.create(
        <Provider store={store}>
            <ModalSignIn
                Animate={Animate}
            />
        </Provider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
