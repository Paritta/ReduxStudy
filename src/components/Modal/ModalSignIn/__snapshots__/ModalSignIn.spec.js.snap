// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSignIn render correctly 1`] = `
.c0 {
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: 5;
  background: rgba(0,0,0,0.3);
}

.c1 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 10px;
  overflow: hidden;
  z-index: 10;
  width: 550px;
  -webkit-animation: 1s eMLfYp;
  animation: 1s eMLfYp;
}

.c2 {
  -webkit-animation: 1s eMLfYp;
  animation: 1s eMLfYp;
}

.c3 {
  font-style: normal;
  font-weight: 400;
  color: white;
  padding: 15px 15px;
  background: palevioletred;
}

.c4 {
  background: white;
  border: 1px solid rgba(0,0,0,0.3);
  padding: 15px;
}

.c5 {
  padding: 30px 0 10px 0;
}

.c6 {
  background-color: #f8f9fa;
  width: 400px;
  height: 40px;
  font-size: 15px;
  border: 2px solid white;
  padding-left: 20px;
  margin-bottom: 10px;
}

.c7 {
  background: palevioletred;
  color: white;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c7:hover {
  background: #f783ac;
  border: 2px solid #f783ac;
}

.c8 {
  padding: 10px 60px;
  margin: 0;
}

.c9 {
  color: white;
  font-size: 1em;
  margin: 5px 0;
  padding: 0.5em 12em 0.5em 1em;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  background: red;
  border: 2px solid red;
}

.c9:hover {
  background: #ff8787;
  border: 2px solid #ff8787;
}

.c10 {
  padding: 0;
  margin: 0 0 0 15px;
}

.c11 {
  color: white;
  font-size: 1em;
  margin: 5px 0;
  padding: 0.5em 12em 0.5em 1em;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  background: royalblue;
  border: 2px solid royalblue;
  padding: 0.5em 11.3em 0.5em 0.5em;
}

.c11:hover {
  background: #4dadf7;
  border: 2px solid #4dadf7;
}

<div>
  <div
    className="c0"
  />
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        로그인
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
        >
          <form
            onSubmit={[Function]}
          >
            <div>
              <input
                className="c6"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="email"
                required={true}
                type="text"
                value=""
              />
            </div>
            <div>
              <input
                className="c6"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <button
              className="c7"
              type="submit"
            >
              로그인
            </button>
            <button
              className="c7"
              onClick={[Function]}
            >
              나가기
            </button>
          </form>
          <div
            className="c8"
          >
            <button
              className="c9"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m20.1 17.5h16.2q0.3 1.5 0.3 2.9 0 4.8-2 8.7t-5.8 5.9-8.7 2.1q-3.5 0-6.6-1.3t-5.5-3.7-3.7-5.4-1.3-6.7 1.3-6.7 3.7-5.4 5.5-3.7 6.6-1.3q6.7 0 11.5 4.4l-4.6 4.5q-2.8-2.6-6.9-2.6-2.8 0-5.3 1.4t-3.9 4-1.4 5.4 1.4 5.4 3.9 4 5.3 1.4q2 0 3.6-0.5t2.7-1.3 1.8-1.9 1.2-1.9 0.5-1.8h-9.8v-5.9z"
                  />
                </g>
              </svg>
              <span
                className="c10"
              >
                Sign in with Google
              </span>
            </button>
            <button
              className="c11"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m29.4 0.3v5.9h-3.5q-1.9 0-2.6 0.8t-0.7 2.4v4.2h6.6l-0.9 6.6h-5.7v16.9h-6.8v-16.9h-5.7v-6.6h5.7v-4.9q0-4.1 2.3-6.4t6.2-2.3q3.3 0 5.1 0.3z"
                  />
                </g>
              </svg>
              <span
                className="c10"
              >
                Sign in with Facebook
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
